docs/testplan/Smartswitch-test-plan.md

```diff
@@ -0,0 +1,876 @@
+#### Sample Output
+```
+On Switch: (shows connection to dpu-4 console and offset is 4 for this case)
+
+root@sonic:/home/cisco# cat /proc/tty/driver/serial 
+serinfo:1.0 driver revision:
+0: uart:16550A port:000003F8 irq:16 tx:286846 rx:6096 oe:1 RTS|DTR|DSR|CD|RI
+1: uart:16550A port:00006000 irq:19 tx:0 rx:0 CTS|DSR|CD
+2: uart:16550A port:000003E8 irq:18 tx:0 rx:0 DSR|CD|RI
+3: uart:16550A port:00007000 irq:16 tx:0 rx:0 CTS|DSR|CD
+**4: uart:16550 mmio:0x94040040 irq:94 tx:0 rx:0
+5: uart:16550 mmio:0x94040060 irq:94 tx:20 rx:68
+6: uart:16550 mmio:0x94040080 irq:94 tx:0 rx:0
+7: uart:16550 mmio:0x940400A0 irq:94 tx:0 rx:0
+8: uart:16550 mmio:0x940400C0 irq:94 tx:0 rx:0
+9: uart:16550 mmio:0x940400E0 irq:94 tx:0 rx:0
+10: uart:16550 mmio:0x94040100 irq:94 tx:0 rx:0
+11: uart:16550 mmio:0x94040120 irq:94 tx:0 rx:0**
+12: uart:16550 mmio:0x94040140 irq:94 tx:0 rx:0 CTS|DSR
+13: uart:16550 mmio:0x94040160 irq:94 tx:0 rx:0 CTS|DSR
+14: uart:16550 mmio:0x94040180 irq:94 tx:0 rx:0 CTS|DSR
+15: uart:16550 mmio:0x940401A0 irq:94 tx:0 rx:0 CTS|DSR
+
+root@sonic:/home/cisco# /usr/bin/picocom -b 115200 /dev/ttyS8
+picocom v3.1
+
+port is        : /dev/ttyS8
+flowcontrol    : none
+baudrate is    : 115200
+parity is      : none
+databits are   : 8
+stopbits are   : 1
+escape is      : C-a
+local echo is  : no
+noinit is      : no
+noreset is     : no
+hangup is      : no
+nolock is      : no
+send_cmd is    : sz -vv
+receive_cmd is : rz -vv -E
+imap is        : 
+omap is        : 
+emap is        : crcrlf,delbs,
+logfile is     : none
+initstring     : none
+exit_after is  : not set
+exit is        : no
+
+Type [C-a] [C-h] to see available commands
+Terminal ready
+
+sonic login: admin
+Password: 
+Linux sonic 6.1.0-11-2-arm64 #1 SMP Debian 6.1.38-4 (2023-08-08) aarch64
+You are on
+  ____   ___  _   _ _  ____
+ / ___| / _ \| \ | (_)/ ___|
+ \___ \| | | |  \| | | |
+  ___) | |_| | |\  | | |___
+ |____/ \___/|_| \_|_|\____|
+
+-- Software for Open Networking in the Cloud --
+
+Unauthorized access and/or use are prohibited.
+All access and/or use are subject to monitoring.
+
+Help:    https://sonic-net.github.io/SONiC/
+
+Last login: Fri Jan 26 21:49:12 UTC 2024 from 169.254.143.2 on pts/1
+admin@sonic:~$ 
+admin@sonic:~$ 
+Terminating...
+Thanks for using picocom
+root@sonic:/home/cisco#
+
+```
+#### Pass/Fail Criteria
+ * Verify Login access is displayed.
+ * cntrl+a and then cntrl+x to come out of the dpu console.
+ 
+ 
+### 1.5 Check midplane ip address between NPU and DPU 
+
+#### Steps
+ * Use command `show ip interface` to get ip addresses 
+ * Get the number of dpu modules from PMON APIs - get_num_modules()
+ * Currently all the interfaces gets static ip address
+ * Use the command `lspci -d 1dd8:1004` to list all the pcie buses for DPUs
+ * Ip adddress mapping to dpu - 169.254.x.2 (switch side interface) and 169.254.x.1 (DPU side interface). where x - bus number in decimal number.
+ * Work in progress - Dymanic assignment of ip address via dhcp
```

Comment:
From the DPU IP assignment design, looks like we shouldn't have the switch side IP interfaces but only a bridge-midplane interface?
@oleksandrivantsiv please help confirm.

Reply:
Correct. The logic described here is incorrect and should be aligned with the IP address assignment HLD.

Reply:
Yes agreed to it. Once the above design is implemented, the test case will be changed as such. This is the temporary solution.

Reply:
This is the community test plan. It should be generic and not based on the specific implementation. I can't approve such PR.

Reply:
Removed the test case as of now and will add it back once the correct design is ready.

Reply:
Please update the sample output. `show ip interface` shows only the `bridge-midplane` address as DPU interfaces don't have directly assigned addresses.
```
Interface        Master    IPv4 address/mask    Admin/Oper    BGP Neighbor    Neighbor IP
---------------  --------  -------------------  ------------  --------------  -------------
bridge-midplane            169.254.200.254/24   up/up         N/A             N/A
docker0                    240.127.1.1/24       up/down       N/A             N/A
eth0                       10.210.26.177/22     up/up         N/A             N/A
lo                         127.0.0.1/16         up/up         N/A             N/A
```